slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -1253,6 +1279,29 @@ void slsDetector::setNumberOfTriggers(int64_t value) {
     shm()->nTriggers = value;
     sendTotalNumFramestoReceiver();
 }
+
+int64_t slsDetector::getNumberOfBurstsFromShm() {
+    return shm()->nBursts; 
+}
+
+int64_t slsDetector::getNumberOfBursts() {
+    int64_t prevVal = shm()->nBursts;
+    int64_t retval = -1;
+    sendToDetector(F_GET_NUM_BURSTS, nullptr, retval);
+    FILE_LOG(logDEBUG1) << "number of bursts :" << retval;
+    shm()->nBursts = retval;
+    if (prevVal != retval) {
```

Comment:
by comparing against shm()->nBursts you can skip using prevVal

Reply:
but i need to update shm()->nBursts before calling sendTotalNumFrametoReceiver().
If I do it before the if condition, I need a new variable.
If I do it inside the if condition, I need to repeat the asignment in the else.

Reply:
done